service/history/replication/executable_history_task.go

```diff
@@ -214,6 +216,41 @@ func (e *ExecutableHistoryTask) HandleErr(err error) error {
 }
 
 func (e *ExecutableHistoryTask) MarkPoisonPill() error {
+	if e.markPoisonPillAttempts >= MarkPoisonPillMaxAttempts {
+
+		events, newRunEvents, err := e.getDeserializedEvents()
```

Comment:
What is the difference between this call and EventSerializer.DeserializeEvents?

Reply:
Maybe we don't need to worry about this deserialize error?

Reply:
`e.getDeserializedEvents()` is to avoid duplicated `DeserializeEvents` to improve some efficiency as when batching different history events tasks, `DeserializeEvents` may be called multiple times.

The handling here is most for logging purpose, when we failed to put task to DLQ, we should try log as much info as possible.